### ISSUE TITLE
Fine-Tune ColBERT in Google Colab-Notebook

### ISSUE THREAD

**Issue by franziskaweber** [stanford-futuredata/ColBERT#228]
Hello,

I am trying to fine-tune the ColBERTv2 pre-trained checkpoint with additional data in a Google Colab-Notebook using the following code:

```
with Run().context(RunConfig(nranks=1)):
  config = ColBERTConfig(bsize=32, lr=3e-06, warmup=None, doc_maxlen=180, dim=128, nway=2, accumsteps=1, use_ib_negatives=False)
  trainer = Trainer(
    triples='triples.tsv',
    queries='queries.tsv',
    collection='documents_train.tsv',
    config=config,
  )

  checkpoint_path = trainer.train(checkpoint=path_to_colbertv2.0)
```

The code is stuck on `#> Starting...`. What can I do to make the training progress?

**Comment by okhat** [comment]
We may need more info on the error to be able to help. The error seems to be masked in the child process because it's running in colab. Basically training forks a new process and the error isn't getting propagated back to the parent, as far as I can tell from this

**Comment by franziskaweber** [comment]
Thank you for your answer!

Do you know if there is any way to get more info on the error in Colab?

**Comment by franziskaweber** [comment]
I figured it out: My triples-file was not in the correct JSONL-format which lead to a trailing data error.

**Comment by roynirmal** [comment]
Hey @franziskaweber  how did you know you had an error? For me i do not see any logs other than an eternal `#> Starting...`

**Comment by roynirmal** [comment]
@franziskaweber did you face a similar issue as this?
https://github.com/stanford-futuredata/ColBERT/issues/254#issue-1904577165 #254

**Comment by franziskaweber** [comment]
Hey @roynirmal, 

I also only saw the eternal `#> Starting...`. But once I fixed the format of my triples-file, everything worked.

**Comment by roynirmal** [comment]
What did you change the format too?  I tried everything (including making it exactly as similar as the ms-marco files)

**Comment by franziskaweber** [comment]
- The file `documents_train.tsv` contains the training documents and has the following format:
0 text0
1 text1
2 text2
...
- The file `queries.tsv` contains the queries and has the following format:
0 query0
1 query1
2 query2
...
- The file `triples.json` contains the training triples and has the following format:
[0, 12692, 48199]
[0, 12693, 17664]
[1, 23783, 21169]
[1, 3796, 17104]
[1, 47897, 30084]
[2, 52241, 54085]
...

The first component is the id of the query, the second one the id of a relevant document and the third one the id of an irrelevant document.

I hope this helps!

**Comment by WeeMan1990** [comment]
> * The file `documents_train.tsv` contains the training documents and has the following format:
>       0 text0
>       1 text1
>       2 text2
>       ...
> 
>     * The file `queries.tsv` contains the queries and has the following format:
>       0 query0
>       1 query1
>       2 query2
>       ...
> 
>     * The file `triples.json` contains the training triples and has the following format:
>       [0, 12692, 48199]
>       [0, 12693, 17664]
>       [1, 23783, 21169]
>       [1, 3796, 17104]
>       [1, 47897, 30084]
>       [2, 52241, 54085]
>       ...
>       The first component is the id of the query, the second one the id of a relevant document and the third one the id of an irrelevant document.
> 
> 
> I hope this helps!
 
I have the exact same issue and have even dummied it down to documents with only 10 lines. I am using the example in the readme for training and have collection/query TSV as mentioned above. 
But when i try to train in colab it stalls on starting. I have tried having the triplets constructed as mentioned above in both TSV and JSONL format. 

In the comment you wrote .json, does extension have any impact and is curly braces needed somewhere in the file?

**Comment by franziskaweber** [comment]
I didn't use any curly braces in the triples-file, just square braces as described above. You can try changing your extension to .json but I don't think that it will have any impact.

I used to following command to create the JSON-file:
`triples_file.write(json.dumps([query_id, rel_doc_id, irrel_doc_id])  + '\n')`
You can try using exactly the same command.

Otherwise, I really can't say what goes wrong in your case. Maybe the problem isn't even the triples-file but something else.
